src/lib/optimizer/strategy/dependent_group_by_reduction_rule.cpp

```diff
@@ -82,44 +82,72 @@ void DependentGroupByReductionRule::_apply_to_plan_without_subqueries(
     auto& aggregate_node = static_cast<AggregateNode&>(*node);
 
     // Early exit: If there are no functional dependencies, we can skip this rule.
-    auto fds = aggregate_node.functional_dependencies();
+    const auto& fds = aggregate_node.functional_dependencies();
     if (fds.empty()) {
       return LQPVisitation::VisitInputs;
     }
 
     // --- Preparation ---
-    // Store a copy of the root's output expressions before applying the rule
+    // Store a copy of the root's output expressions before applying the rule.
     const auto root_output_expressions = lqp_root->output_expressions();
     // Also store a copy of the aggregate's output expressions to verify the output column order later on.
     const auto initial_aggregate_output_expressions = aggregate_node.output_expressions();
 
     // Gather group-by columns
     const auto fetch_group_by_columns = [&aggregate_node]() {
-      ExpressionUnorderedSet group_by_columns(aggregate_node.aggregate_expressions_begin_idx + 1);
+      auto group_by_columns = ExpressionUnorderedSet{aggregate_node.aggregate_expressions_begin_idx + 1};
       auto node_expressions_iter = aggregate_node.node_expressions.cbegin();
       std::copy(node_expressions_iter, node_expressions_iter + aggregate_node.aggregate_expressions_begin_idx,
                 std::inserter(group_by_columns, group_by_columns.end()));
       return group_by_columns;
     };
     auto group_by_columns = fetch_group_by_columns();
 
+    // Get a sorted list of ColumnIDs from an FD's set of determinants.
+    const auto get_column_ids = [&](const auto& determinants) {
+      auto column_ids = std::vector<ColumnID>{};
+      column_ids.reserve(determinants.size());
+      std::transform(determinants.cbegin(), determinants.cend(), std::back_inserter(column_ids),
```

Comment:
I don't see transform here being better to read than a range-based for loop. I am not asking you to change that, but is there a particular reason you are using transform?

Reply:
No particular reason. Think a for loop does not look that weird for some ppl, gonna change that.